pkg/apis/apps/v1alpha1/default.go

```diff
@@ -227,6 +229,14 @@ func SetDefaultsYurtStaticSet(obj *YurtStaticSet) {
 	if podSpec != nil {
 		SetDefaultPodSpec(podSpec)
 	}
+
+	// use YurtStaticSet name and namespace to replace name and namespace in template metadata
+	if !reflect.DeepEqual(obj.Spec, reflect.Zero(reflect.TypeOf(obj.Spec)).Interface()) {
+		if _, ok := reflect.TypeOf(obj.Spec).FieldByName("Template"); ok {
+			obj.Spec.Template.Name = obj.Name
+			obj.Spec.Template.Namespace = obj.Namespace
+		}
+	}
```

Comment:
Line234~239 can be optimized as following:
```
obj.Spec.Template.Name = obj.Name
obj.Spec.Template.Namespace = obj.Namespace
```

Reply:
That's what I used to write, but @fengshunli suggest Obj.Spec adds empty finger judgment, so whether to modify it?

Reply:
> That's what I used to write, but @fengshunli suggest Obj.Spec adds empty finger judgment, so whether to modify it?

@rambohe-ch WDYT？

Reply:
@fengshunli @JameKeal I think we don't need to add condition for this code snippet, because `obj.Spec.Template` field must be present.

Reply:
@fengshunli Take K8s code as example, you can check the default setting for daemonset creation in the following link: 
https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/apps/daemonset/strategy.go#L79-L81

Reply:
> @fengshunli @JameKeal I think we don't need to add condition for this code snippet, because `obj.Spec.Template` field must be present.

I think so.